modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/Tabs.js

```diff
@@ -37,6 +37,27 @@ export default function Tabs({
 		tabsItems.find((item) => item.active)
 	);
 
+	function isRenderNode(element) {
+	  if (element.nodeName == "DIV" && element.innerText == "") {
+        const id = element.getAttribute("id");
+        if (id && id.startsWith("__RENDER__")) {
+          return id;
+      }
+    }
+	  return null;
+  }
+
+
+	useEffect(() => {
+    panels.map((panel) => {
+        const panelDocument = new DOMParser().parseFromString(panel, "text/html");
+        panelDocument.body.querySelectorAll("div div[id^=__RENDER__]").forEach((c) => {
+          const id = isRenderNode(c);
```

Comment:
when inline rendering is on, `panel` will not only contain the container markup for hydration but also the script that calls render.
As Tabs is using `dangerouslySetInnerHTML` for the panels, these scripts [won't run ](https://suncommander.medium.com/how-to-inject-scripts-using-dangerouslysetinnerhtml-on-client-side-rendering-973037cc06b7).

So, this effect can also be in charge of handling the execution of these as they can be easily detected

Reply:
another caveat is that we probably don't want to run the hydration for all the potential nested tags but only to the first-level ones, in order to guarantee proper hydration order in the hierarchy. This is something I have not checked